src/lib.rs

```diff
@@ -1168,6 +1169,25 @@ impl Builder {
 
     /// Generate the Rust bindings using the options built up thus far.
     pub fn generate(mut self) -> Result<Bindings, ()> {
+        // Add any extra arguments from the environment to the clang command line.
+        if let Some(extra_clang_args) = std::env::var("BINDGEN_EXTRA_CLANG_ARGS").ok() {
+            // Try to parse it with shell quoting. If we fail, make it one single big argument.
+            let (strings, parse_error) = {
```

Comment:
Can't this just use `shlex::split`? If not, why not?

Reply:
It can't because of error handling. From the docs:

> An input string is erroneous if it ends while inside a quotation or right after an unescaped backslash. Since Iterator does not have a mechanism to return an error, if that happens, Shlex just throws out the last token, ends the iteration, and sets 'had_error' to true; best to check it after you're done iterating.

So if the object is consumed or has its type erased, there is no way to check for errors.

This code is the least ugly way I could come up with to allow error checking.